atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/StreamStoreRemappingNextTableToSweepProviderImpl.java

```diff
@@ -17,39 +17,62 @@
 package com.palantir.atlasdb.sweep.priority;
 
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.schema.stream.StreamTableType;
 import com.palantir.atlasdb.transaction.api.Transaction;
 
 public class StreamStoreRemappingNextTableToSweepProviderImpl implements NextTableToSweepProvider {
+    private static final long MILLIS_TO_WAIT_BEFORE_SWEEP_VALUE_TABLE = TimeUnit.HOURS.toMillis(1);
+
     private NextTableToSweepProviderImpl delegate;
-    private boolean hasRemappedStreamStoreValueTable = false;
-    private TableReference previousStreamStoreValueTable = null;
+    private State state = State.DEFAULT;
```

Comment:
does this logic need to be stateful? seems like you could do:
- if last values table sweep time > last index table sweep time:
   values priority = 0
   index priority = priority we'd normally give the values table
- else if last index table sweep < 2 hours ago:
   values priority = 0
   index priority = 0
- else (last index sweep > 2 hours ago and last values sweep < last index sweep)
   values priority = normal
   index priority = 0

Reply:
just saw that @hsaraogi suggested a similar thing as well